__tests__/supertest.js

```diff
@@ -0,0 +1,417 @@
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- attributes should be an array `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('attributes', '5000');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- attributes should be an array of objects `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('attributes', ['5000']);
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- planter_identifier is required `, function (done) {
+      const request_object = new RequestObject();
+      request_object.delete_property('planter_identifier');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- timestamp is required `, function (done) {
+      const request_object = new RequestObject();
+      request_object.delete_property('timestamp');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- should be of type unixtimestamp in seconds `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('timestamp', 'ashbfhd');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    describe('Raw capture should be added sucessfully', () => {
+      after(function (done) {
+        seed.delete_inserted_raw_capture().then(() => done());
```

Comment:
I am wondering should we worry about deleting the created entry after the test. The convenience it probably offers is a static payload so that it can be run multiple times. I am wondering if we should generate new random uuid (for each new api invocation) and leave the test data in the db that was created due to the api invocation.

The reason for my thought is, for api tests, we could treat the backend storage as a black box and the tests  only interact with the api endpoints. What do you think? Do you see any downside for this strategy?

Reply:
Not really, I guess that is fine. I'll make changes to this